### DISCUSSION TITLE
net2net-cert question

### DISCUSSION THREAD

**Discussion by FrontWhite** [strongswan/strongswan#1923] [Q&A]
enviroment:
192.168.5.0/24 -- | 192.168.41.71 | === | 192.168.45.108 | -- 192.168.1.0/24
  moon-net          moon                 sun           sun-net
  
My configuration：
moon：connections {
	ipsec1 {
		local_addrs = 192.168.41.71
		remote_addrs = 192.168.45.108
		version = 2
		aggressive = no
		mobike = no
		proposals = aes128-sha1-modp1024
		rekey_time = 3600s
		local {
			auth = pubkey
			certs = moonCert.pem
			id = "C=CN, O=strongswan, CN=192.168.41.71"
		}
		remote {
			auth = pubkey
			id = "C=CN, O=strongswan, CN=192.168.45.108"
		}
		children {
			ipsec1_child {
				local_ts = 192.168.5.0/24
				remote_ts = 192.168.1.0/24
				rekey_time = 3600s
				esp_proposals = aes128-sha1-modp1024
				start_action = none
				mode = tunnel
			}
		}
	}
}
sun:
connections {
	ipsec1 {
		local_addrs = 192.168.45.108
		remote_addrs = 192.168.41.71
		version = 2
		aggressive = no
		mobike = no
		proposals = aes128-sha1-modp1024
		rekey_time = 3600s
		local {
			auth = pubkey
			certs = sunCert.pem
			id = "C=CN, O=strongswan, CN=192.168.45.108"
		}
		remote {
			auth = pubkey
			id = "C=CN, O=strongswan, CN=192.168.41.71"
		}
		children {
			ipsec1_child {
				local_ts = 192.168.1.0/24
				remote_ts = 192.168.5.0/24
				rekey_time = 3600s
				esp_proposals = aes128-sha1-modp1024
				start_action = none
				mode = tunnel
			}
		}
	}
}

swanctl --load-all
loaded certificate from '/etc/swanctl/x509/moonCert.pem'
loaded certificate from '/etc/swanctl/x509ca/strongswanCert.pem'
loaded RSA key from '/etc/swanctl/private/moonKey.pem'
no authorities found, 0 unloaded
no pools found, 0 unloaded
loaded connection 'ipsec1'
successfully loaded 1 connections, 0 unloaded

 ipsec up ipsec1
initiating IKE_SA ipsec1[7] to 192.168.45.108
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 192.168.41.71[500] to 192.168.45.108[500] (336 bytes)
received packet: from 192.168.45.108[500] to 192.168.41.71[500] (369 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
received 1 cert requests for an unknown ca
sending cert request for "C=CN, O=strongSwan, CN=strongSwan Root CA"
authentication of 'C=CN, O=strongswan, CN=192.168.41.71' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
sending end entity cert "C=CN, O=strongswan, CN=192.168.41.71"
establishing CHILD_SA ipsec1_child{6}
generating IKE_AUTH request 1 [ IDi CERT N(INIT_CONTACT) CERTREQ IDr AUTH SA TSi TSr N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
splitting IKE message (1452 bytes) into 2 fragments
generating IKE_AUTH request 1 [ EF(1/2) ]
generating IKE_AUTH request 1 [ EF(2/2) ]
sending packet: from 192.168.41.71[500] to 192.168.45.108[500] (1248 bytes)
sending packet: from 192.168.41.71[500] to 192.168.45.108[500] (272 bytes)
received packet: from 192.168.45.108[500] to 192.168.41.71[500] (76 bytes)
parsed IKE_AUTH response 1 [ N(AUTH_FAILED) ]
received AUTHENTICATION_FAILED notify error
establishing connection 'ipsec1' failed

**Question: I must also load the sun's certificate configuration on the moon side, the sun side loads the moon's certificate configuration to connect successfully, otherwise it will fail as above, I see that the "strongSwan Documentation" does not mention the need to load the peer certificate, is there a problem with my certificate, or what step is wrong?**

 swanctl --load-all
loaded certificate from '/etc/swanctl/x509/sunCert.pem'
loaded certificate from '/etc/swanctl/x509/moonCert.pem'
loaded certificate from '/etc/swanctl/x509ca/strongswanCert.pem'
loaded RSA key from '/etc/swanctl/private/sunKey.pem'
loaded RSA key from '/etc/swanctl/private/moonKey.pem'
no authorities found, 0 unloaded
no pools found, 0 unloaded
loaded connection 'ipsec1'
successfully loaded 1 connections, 0 unloaded

ipsec up ipsec1
initiating IKE_SA ipsec1[10] to 192.168.45.108
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 192.168.41.71[500] to 192.168.45.108[500] (336 bytes)
received packet: from 192.168.45.108[500] to 192.168.41.71[500] (369 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
selected proposal: IKE:AES_CBC_128/HMAC_SHA1_96/PRF_HMAC_SHA1/MODP_1024
received 1 cert requests for an unknown ca
sending cert request for "C=CN, O=strongSwan, CN=strongSwan Root CA"
authentication of 'C=CN, O=strongswan, CN=192.168.45.108' (myself) with RSA_EMSA_PKCS1_SHA2_256 successful
sending end entity cert "C=CN, O=strongswan, CN=192.168.45.108"
establishing CHILD_SA ipsec1_child{7}
generating IKE_AUTH request 1 [ IDi CERT N(INIT_CONTACT) CERTREQ IDr AUTH SA TSi TSr N(MULT_AUTH) N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
splitting IKE message (1452 bytes) into 2 fragments
generating IKE_AUTH request 1 [ EF(1/2) ]
generating IKE_AUTH request 1 [ EF(2/2) ]
sending packet: from 192.168.41.71[500] to 192.168.45.108[500] (1248 bytes)
sending packet: from 192.168.41.71[500] to 192.168.45.108[500] (272 bytes)
received packet: from 192.168.45.108[500] to 192.168.41.71[500] (1248 bytes)
parsed IKE_AUTH response 1 [ EF(1/2) ]
received fragment #1 of 2, waiting for complete IKE message
received packet: from 192.168.45.108[500] to 192.168.41.71[500] (144 bytes)
parsed IKE_AUTH response 1 [ EF(2/2) ]
received fragment #2 of 2, reassembled fragmented IKE message (1324 bytes)
parsed IKE_AUTH response 1 [ IDr CERT AUTH SA TSi TSr ]
received end entity cert "C=CN, O=strongswan, CN=192.168.41.71"
no issuer certificate found for "C=CN, O=strongswan, CN=192.168.41.71"
  issuer is "C=CN, O=strongSwan, CN=strongSwan Root CA"
  using trusted certificate "C=CN, O=strongswan, CN=192.168.41.71"
authentication of 'C=CN, O=strongswan, CN=192.168.41.71' with RSA_EMSA_PKCS1_SHA2_256 successful
IKE_SA ipsec1[10] established between 192.168.41.71[C=CN, O=strongswan, CN=192.168.45.108]...192.168.45.108[C=CN, O=strongswan, CN=192.168.41.71]
scheduling rekeying in 3564s
maximum IKE_SA lifetime 3924s
[947]:ipsec status === 3
[process_i,1474]: ipsec status === 1
selected proposal: ESP:AES_CBC_128/HMAC_SHA1_96/NO_EXT_SEQ
CHILD_SA ipsec1_child{7} established with SPIs ce09dcd3_i cd108905_o and TS 192.168.5.0/24 === 192.168.1.0/24
[select_and_install,840]: ipsec status === 3
connection 'ipsec1' established successfully
#

**Comment by tobiasbrunner** [comment]
> is there a problem with my certificate

Could be. You should check what the actual error is on the responder (the log you provided doesn't help as it only shows that an error notify was received).

**Reply by FrontWhite** [reply]
I still don't quite understand the reason. The following is my certificate configuration process. I configured the certificate according to the document：

mkdir tmp/ipsec_cert
pki --gen --outform pem > /tmp/ipsec_cert/strongswanKey.pem
pki --self --ca --lifetime 3652 --in /tmp/ipsec_cert/strongswanKey.pem \
           --dn "C=CN, O=strongSwan, CN=strongSwan Root CA" \
           --outform pem > /etc/swanctl/x509ca/strongswanCert.pem
pki --gen --outform pem > /etc/swanctl/private/moonKey.pem
pki --req --type priv --in /etc/swanctl/private/moonKey.pem \
          --dn "C=CN, O=strongswan, CN=192.168.41.71" \
          --san 192.168.41.71 --outform pem > /tmp/ipsec_cert/moonReq.pem
pki --issue --cacert /etc/swanctl/x509ca/strongswanCert.pem --cakey /tmp/ipsec_cert/strongswanKey.pem \
            --type pkcs10 --in /tmp/ipsec_cert/moonReq.pem --serial 01 --lifetime 1826 \
            --outform pem > /etc/swanctl/x509/moonCert.pem

>You should check what the actual error is on the responder

What additional log level do I need to turn on？I don't see any useful information in the /var/log/charon.log.

**Reply by tobiasbrunner** [reply]
No additional log levels, you have to check the log on the right host (the initiator only reports that the responder rejected the authentication, so that's not useful to find out why this happened). But for debugging, check the recommended log levels in #196.
